tsdb/index/postings_test.go

```diff
@@ -1282,3 +1283,18 @@ func BenchmarkListPostings(b *testing.B) {
 		})
 	}
 }
+
+func TestMemPostings_PostingsForLabelMatchingHonorsContextCancel(t *testing.T) {
+	memP := NewMemPostings()
+
+	for i := 1; i <= 10000; i++ {
+		memP.Add(storage.SeriesRef(i), labels.FromStrings("__name__", fmt.Sprintf("%4d", i)))
+	}
+
+	ctx := &testutil.MockContextErrAfter{FailAfter: 50}
+	p := memP.PostingsForLabelMatching(ctx, "__name__", func(string) bool {
+		return true
+	})
+	require.Error(t, p.Err())
+	require.GreaterOrEqual(t, ctx.Count(), uint64(50))
```

Comment:
It doesn't tell us that interrupted the iteration. I would add another assertion to ensure the count is less than X (where X is < 10000). You can also be very specific and just assert that `ctx.Count()` is exactly 100, given we expect to check it every 100 iterations. Then 100 could be a constant, so that we use it both here and in the actual implementation.